syncer/syncer.go

```diff
@@ -1799,7 +1798,7 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 
 	s.tctx.L().Info(annotate, zap.String("event", "query"), zap.String("source", source), zap.Strings("ddls", needHandleDDLs), zap.ByteString("raw statement", ev.Query), zap.Bool("in-sharding", needShardingHandle), zap.Stringer("start location", startLocation), zap.Bool("is-synced", synced), zap.Int("unsynced", remain))
 
-	if err := s.flushCheckPoints(); err != nil {
+	if err := s.flushJobs(); err != nil {
```

Comment:
It seems we can remove `s.jobWg.Wait()` in L1856 now? (or even we can remove `s.jobWg.Wait()` in L723? but we may still just keep it, the shard mechanism make this not uniform, 😢 )